src/result_processor.c

```diff
@@ -649,13 +649,22 @@ ResultProcessor *RPLoader_New(RLookup *lk, const RLookupKey **keys, size_t nkeys
   sc->lk = lk;
   sc->base.Next = rploaderNext;
   sc->base.Free = rploaderFree;
-  sc->base.name = "Loader";
+  sc->base.type = RP_LOADER;
   return &sc->base;
 }
 
+static char *RPTypeLookup[RP_MAX] = {
+  "Index", "Loader", "Scorer", "Sorter", "Pager/Limiter",
+  "Grouper", "Projector", "Filter", "Profile", "Network"};
+
+const char *RPTypeToString(ResultProcessorType type) {
```

Comment:
Should we check whether the enum value passed to RPTypeToString is legal (e.g., <RP_MAX)